Panic in renderText

Source:
```md
test

My favorite search engine is [Duck Duck Go](https://duckduckgo.com).

![The San Juan Mountains are beautiful!](https://mdg.imgix.net/assets/images/san-juan-mountains.jpg?auto=format&fit=clip&q=40&w=1080 "San Juan Mountains")

Here's our logo (hover to see the title text):

Inline-style: 
![alt text](https://github.com/adam-p/markdown-here/raw/master/src/common/images/icon48.png "Logo Title Text 1")

Reference-style: 
![alt text][logo]

[logo]: https://github.com/adam-p/markdown-here/raw/master/src/common/images/icon48.png "Logo Title Text 2"

ok.
```

returns
```go
panic: interface conversion: ast.Node is *ast.String, not *ast.Text

goroutine 1 [running]:
github.com/stephenafamo/goldmark-pdf.(*nodeRederFuncs).renderText(0xc0002cc7ec?, 0xd2f3a0?, {0xc0003c4000?, 0x0?, 0x0?}, {0xd3dde0?, 0xc0000df3b0?}, 0x0?)
        G:/Go/pkg/mod/github.com/stephenafamo/goldmark-pdf@v0.2.0/renderer_funcs.go:138 +0xd0
github.com/stephenafamo/goldmark-pdf.(*renderer).Render.func2({0xd3dde0, 0xc0000df3b0}, 0x80?)
        G:/Go/pkg/mod/github.com/stephenafamo/goldmark-pdf@v0.2.0/renderer.go:103 +0xb6
github.com/yuin/goldmark/ast.walkHelper({0xd3dde0, 0xc0000df3b0}, 0xc000515d08)
        G:/Go/pkg/mod/github.com/yuin/goldmark@v1.5.3/ast/ast.go:492 +0x34
github.com/yuin/goldmark/ast.walkHelper({0xd3dba0, 0xc000036e00}, 0xc000515d08)
        G:/Go/pkg/mod/github.com/yuin/goldmark@v1.5.3/ast/ast.go:498 +0x8e
github.com/yuin/goldmark/ast.walkHelper({0xd3d180, 0xc0000defc0}, 0xc000515d08)
        G:/Go/pkg/mod/github.com/yuin/goldmark@v1.5.3/ast/ast.go:498 +0x8e
github.com/yuin/goldmark/ast.Walk(...)
        G:/Go/pkg/mod/github.com/yuin/goldmark@v1.5.3/ast/ast.go:487
github.com/stephenafamo/goldmark-pdf.(*renderer).Render(0xc000356440, {0xd2ede0, 0xc00008a038}, {0xc0003c4000, 0x230, 0x380}, {0xd3d180, 0xc0000defc0})
        G:/Go/pkg/mod/github.com/stephenafamo/goldmark-pdf@v0.2.0/renderer.go:98 +0x632
github.com/yuin/goldmark.(*markdown).Convert(0xc000356480, {0xc0003c4000, 0x230, 0x380}, {0xd2ede0, 0xc00008a038}, {0x0, 0x0, 0x0})
```

--- a/renderer_funcs.go
+++ b/renderer_funcs.go
@@ -141,17 +141,25 @@ func (r *nodeRederFuncs) renderText(w *Writer, source []byte, node ast.Node, ent
 		return ast.WalkContinue, nil
 	}
 
-	// Soft line breaks (a bare newline in the source between inline content)
-	// aren't part of the segment value — goldmark exposes them via a flag.
-	// CommonMark renders them as a space when reflowing inline text, so we
-	// append one. Without this, content like "[link](url)\nnext word" runs
-	// the two together as "linknext".
-	n := node.(*ast.Text)
-	text := string(n.Segment.Value(source))
-	if n.SoftLineBreak() {
-		text += " "
+	// Both KindText and KindString are registered to this func. *ast.String
+	// is emitted by extensions (e.g. Typographer's smart-quote/dash/ellipsis
+	// substitutions) and stores its bytes directly on Value rather than via
+	// a source Segment.
+	switch n := node.(type) {
+	case *ast.Text:
+		// Soft line breaks (a bare newline in the source between inline
+		// content) aren't part of the segment value — goldmark exposes them
+		// via a flag. CommonMark renders them as a space when reflowing
+		// inline text, so we append one. Without this, content like
+		// "[link](url)\nnext word" runs the two together as "linknext".
+		text := string(n.Segment.Value(source))
+		if n.SoftLineBreak() {
+			text += " "
+		}
+		w.WriteText(text)
+	case *ast.String:
+		w.WriteText(string(n.Value))
 	}
-	w.WriteText(text)
 
 	return ast.WalkContinue, nil
 }

--- a/renderer_funcs_test.go
+++ b/renderer_funcs_test.go
@@ -1,0 +1,34 @@
+package pdf
+
+import (
+	"bytes"
+	"testing"
+
+	"github.com/yuin/goldmark"
+	"github.com/yuin/goldmark/extension"
+)
+
+// renderText is registered for both KindText and KindString but
+// unconditionally cast to *ast.Text, panicking when an extension emits
+// *ast.String (e.g. Typographer's smart-quote/dash substitutions).
+func TestRender_AstStringDoesNotPanic(t *testing.T) {
+	md := goldmark.New(
+		goldmark.WithExtensions(extension.Typographer),
+		goldmark.WithRenderer(New(
+			WithPDF(&MockPdf{pageWidth: 600, pageHeight: 800, leftMargin: 50, rightMargin: 50}),
+			// Inbuilt fonts skip the Google-fonts network fetch.
+			WithHeadingFont(FontHelvetica),
+			WithBodyFont(FontHelvetica),
+			WithCodeFont(FontCourier),
+		)),
+	)
+
+	// Typographer rewrites the dashes, ellipsis, and quotes into *ast.String
+	// nodes interleaved with the surrounding *ast.Text.
+	const sample = "Hello -- world... and \"quotes\".\n"
+
+	var buf bytes.Buffer
+	if err := md.Convert([]byte(sample), &buf); err != nil {
+		t.Fatalf("convert: %v", err)
+	}
+}

